File: tools/ghidra/FindClientClassSymbols.py
from ghidra.program.model.symbol import SymbolType


def print_matches(needle, limit=40):
    needle_lower = needle.lower()
    print("=== SYMBOL MATCHES: %s ===" % needle)

    symbol_table = currentProgram.getSymbolTable()
    matches = []
    it = symbol_table.getAllSymbols(True)
    while it.hasNext():
        sym = it.next()
        name = sym.getName(True)
        if needle_lower in name.lower():
            matches.append(sym)

    matches.sort(key=lambda s: s.getAddress().getOffset())
    if not matches:
        print("  <none>")
        return

    for sym in matches[:limit]:
        print(
            "  %s  %-12s  %s"
            % (sym.getAddress(), str(sym.getSymbolType()), sym.getName(True))
        )


def main():
    print("PROGRAM: %s" % currentProgram.getExecutablePath())
    print("LANGUAGE: %s" % currentProgram.getLanguageID())
    print("IMAGE BASE: %s" % currentProgram.getImageBase())

    needles = [
        "C_BaseEntity",
        "C_BaseAnimating",
        "C_BaseFlex",
        "C_BaseCombatCharacter",
        "CTerrorPlayer",
        "C_TerrorPlayer",
        "ClientRagdoll",
        "SetupBones",
        "UpdateClientSideAnimation",
        "ShouldInterpolate",
        "vftable",
    ]

    for needle in needles:
        print_matches(needle)


main()

File: tools/ghidra/FindClientClassSymbols.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.Iterator;
import java.util.List;

import ghidra.app.script.GhidraScript;
import ghidra.program.model.symbol.Symbol;
import ghidra.program.model.symbol.SymbolTable;

public class FindClientClassSymbols extends GhidraScript {

	private void printMatches(String needle, int limit) {
		String needleLower = needle.toLowerCase();
		println("=== SYMBOL MATCHES: " + needle + " ===");

		SymbolTable symbolTable = currentProgram.getSymbolTable();
		List<Symbol> matches = new ArrayList<>();
		Iterator<Symbol> iterator = symbolTable.getAllSymbols(true);
		while (iterator.hasNext()) {
			Symbol symbol = iterator.next();
			String name = symbol.getName(true);
			if (name != null && name.toLowerCase().contains(needleLower)) {
				matches.add(symbol);
			}
		}

		Collections.sort(matches, Comparator.comparingLong(s -> s.getAddress().getOffset()));
		if (matches.isEmpty()) {
			println("  <none>");
			return;
		}

		int count = 0;
		for (Symbol symbol : matches) {
			printf("  %s  %-12s  %s\n",
				symbol.getAddress(),
				String.valueOf(symbol.getSymbolType()),
				symbol.getName(true));
			if (++count >= limit) {
				break;
			}
		}
	}

	@Override
	protected void run() throws Exception {
		println("PROGRAM: " + currentProgram.getExecutablePath());
		println("LANGUAGE: " + currentProgram.getLanguageID());
		println("IMAGE BASE: " + currentProgram.getImageBase());

		String[] needles = new String[] {
			"C_BaseEntity",
			"C_BaseAnimating",
			"C_BaseFlex",
			"C_BaseCombatCharacter",
			"CTerrorPlayer",
			"C_TerrorPlayer",
			"ClientRagdoll",
			"SetupBones",
			"UpdateClientSideAnimation",
			"ShouldInterpolate",
			"vftable"
		};

		for (String needle : needles) {
			printMatches(needle, 40);
		}
	}
}
